guides/plugins/plugins/framework/store-api/add-store-api-route.md

```diff
@@ -0,0 +1,297 @@
+we are returning a JSON object. Within the JSON content, we have three properties annotated with `@OA\Property`.
+The first property is the amount of entities we retrieved. The next property contains the aggregations of our criteria.
+
+Finally, we have our retrieved entities, using an `@OA\Items` annotation which references to `#/components/schemas/`
+and the technical name of our entity, so in our case `#/components/schemas/swag_example`.
+This is used to generate the schema according to our definition.
+
+### Route response
+
+After we have created our route, we need to create the mentioned `ExampleRouteResponse`. This class should extend
+from `Shopware\Core\System\SalesChannel\StoreApiResponse`. In this class we have a property `$object` of type
+`Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult`. The class constructor has one argument 
+`EntitySearchResult`, which we passed to it in our `ExampleRoute`. The constructor calls the parent constructor with the 
+parameter `$object` which sets the value for our object property. Finally, we add a method `getExamples` in which we 
+return our entity collection that we got from the object.
+
+{% code title="<plugin root>/src/Core/Content/Example/SalesChannel/ExampleRouteResponse.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Core\Content\Example\SalesChannel;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Shopware\Core\System\SalesChannel\StoreApiResponse;
+use Swag\BasicExample\Core\Content\Example\ExampleCollection;
+
+class ExampleRouteResponse extends StoreApiResponse
+{
+    /**
+     * @var EntitySearchResult
+     */
+    protected $object;
+
+    public function __construct(EntitySearchResult $object)
+    {
+        parent::__construct($object);
+    }
+
+    public function getExamples(): ExampleCollection
+    {
+        /** @var ExampleCollection $collection */
+        $collection = $this->object->getEntities();
+
+        return $collection;
+    }
+}
+```
+{% endcode %}
+
+## Register route
+
+The last thing we need to do now is to tell Shopware how to look for new routes in our plugin.
+This is done with a `routes.xml` file at `<plugin root>/src/Resources/config/` location.
+Have a look at the official [Symfony documentation](https://symfony.com/doc/current/routing.html) about routes and how they are registered.
+
+{% code title="<plugin root>/src/Resources/config/routes.xml" %}
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<routes xmlns="http://symfony.com/schema/routing"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://symfony.com/schema/routing
+        https://symfony.com/schema/routing/routing-1.0.xsd">
+  
+    <import resource="../../Core/**/*Route.php" type="annotation" />
+</routes>
+```
+{% endcode %}
+
+## Check route in Swagger
```

Comment:
I hate Swagger, and most developers do as well 

If I am not implementing the billions of Swagger Annotations because I am lazy, my only chance is the `bin/console debug:router` command, to check, if my routes were loaded

Please add two or three sentences and a reference to it https://symfony.com/doc/current/routing.html#debugging-routes